airbyte-integrations/connectors/source-kapiche-export-api/source_kapiche_export_api/source.py

```diff
@@ -0,0 +1,356 @@
+
+
+# Basic full refresh stream
+class KapicheExportApiStream(HttpStream, ABC):
+    """
+    Abstract Stream class for kapiche-export-api connector.
+    All streams from airbyte must inherit from this as a base class.
+    """
+    
+    def __init__(self, authenticator: HttpAuthenticator = None):
+        super().__init__(authenticator)
+        self._url_base = ''
+
+    @property
+    def url_base(self) -> str:
+        """
+        :return: URL base for the  API endpoint e.g: if you wanted to hit https://myapi.com/v1/some_entity then this should return "https://myapi.com/v1/"
+        """
+        return self._url_base
+
+    @url_base.setter
+    def url_base(self, value:str):
+        self._url_base =  value
+
+    
+
+
+class ExportDataGet(KapicheExportApiStream, ABC):
+    """
+    TODO: Change class name to match the table/data source this stream corresponds to.
+    """
+
+    primary_key = None
+    cursor_field = 'document_id__'
+    
+    # needed to instantiate the class
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        if x := response.headers.get('Kapiche-next-document-id'):
+            return { self.cursor_field: x}
+        else:
+            return None
+
+    def __init__(
+        self,
+        export_uuid: str,
+        authenticator: HttpAuthenticator,
+        url: str = "",
+        name: str = None,
+    ):
+
+        self._session = requests.Session()
+        self._authenticator = authenticator
+        self.export_uuid = export_uuid
+        self._cursor_value = 1
+        self._url_base = url
+        self._name = name
+    
+    @property
+    def name(self) -> str:
+        """
+        :return: Stream name. By default this is the implementing class name, but it can be overridden as needed.
+        """
+        return f'kapiche-{self._name}'
+    
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]):
+        if current_stream_state.get(self.cursor_field):
+            if int(current_stream_state[self.cursor_field]) < int(latest_record[self.cursor_field]):
+                return {self.cursor_field: latest_record[self.cursor_field]}
+            else:
+                return {self.cursor_field: current_stream_state[self.cursor_field]}
+
+        return self.state
+
+    def path(self, *args, **kwargs) -> str:
+        """
+        Returns the URL path for the endpoint. 
+        """
+        return f"document-queries/"
+    
+    @property
+    def cursor_field(self) -> str:
+        """
+        This field's presence tells the framework this in an incremental stream. Required for incremental.
+
+        :return str: The name of the cursor field.
+        """
+        return 'document_id__'
+   
+    @property
+    def state(self) -> Mapping[str, Any]:
+        return {self.cursor_field: self._cursor_value}
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]):
```

Comment:
Would this be:

```suggestion
    def state(self, value: Mapping[str, int]):
```

?